webiste/integrations: update roundcube doc (#16753)
goauthentik/authentik: website/integrations/chat-communication-collaboration/roundcube/index.md

## What is Roundcube

> Roundcube is a browser-based multilingual IMAP client with an application-like user interface. It provides full functionality you expect from an email client, including MIME support, address book, folder manipulation, message searching and spell checking.
>
> -- https://roundcube.net

## Preparation

The following placeholders are used in this guide:

- `authentik.company` is the FQDN of the authentik installation.
- `roudcube.company` is the FQDN of the Roundcube installation.

:::note

> This document describes how to use Roundcube's oauth support with authentik to automatically sign into an email account.

Reviewer: What else would we use this for? I don't feel like this sentence is needed.

> The mail server must support XOAUTH2 for both SMTPD and IMAP/POP. Postfix SMTP server can also use Dovecot for authentication which provides Postfix with XOAUTH2 capability without configuring it separately.

Reviewer: I feel like a few things should be made clear: This is a roundcube guide, why are we mentioning alternatives here? What exactly is Postfix and what role does it play in all this? It's mentioned here, then never again. At the minimum, sentencing could be cleaned up to this, which might make it slightly more readable:

```suggestion
The mail server must support XOAUTH2 for both SMTPD and IMAP/POP. A Postfix SMTP server can use Dovecot for authentication, which allows XOAUTH2 support in Postfix without requiring separate configuration.
```

Even with that being said, I'd like to see what a Postfix SMTP server is and what is Dovcot. Perhaps we can include some text in parenthesis explaining the relation or straight up removing this bit?

Reviewer: Ok, after scrolling down I do see the Dovecot configuration block, but it left a question mark in my brain for a bit. Mabye it's just me?

Author: I've added a block at the beginning which explains the use of dovecot and postfix.

### Create property mappings

> 1. Log in to authentik as an administrator, and open the authentik Admin interface.

Reviewer: By the looks if it, this isn't the case at the moment...

<img width="220" height="165" alt="image" src="https://github.com/user-attachments/assets/ddbf6d12-c513-4bb4-ae62-0feb0a30792c" />

With comma: https://github.com/search?q=repo%3Agoauthentik%2Fauthentik+%22Log+in+to+authentik+as+an+administrator%2C+and+open+the+authentik+Admin+interface.%22&type=code
Without: https://github.com/search?q=repo%3Agoauthentik%2Fauthentik+%22Log+in+to+authentik+as+an+administrator+and+open+the+authentik+Admin+interface.%22&type=code

However, I'd be open to discussing the potential addition of this comma in a later PR as this change would affect all integrations

Author: I've removed both. Weirdly, I have a memory of being corrected on this but it appears that without a comma is correct.

            "family_name": "",
            "preferred_username": request.user.username,
            "nickname": request.user.username,
            "active": True,
        }
        ```

### Create an application and provider in authentik

> 1. Log in to authentik as an administrator, and open the authentik Admin interface.

Reviewer: Same comment as line 36

    - Note the **Client ID**, **Client Secret**, and **slug** values because they will be required later.
    - Set a `Strict` redirect URI to `https://roundcube.company/index.php?\_task=settings&\_action=plugin.oauth_redirect`.
    - Select any available signing key.
    - Under **Advanced protocol settings**:

>         - Add the `dovecotprofile` and `authentik default OAuth Mapping: OpenID 'offline_access'` scopes to selected scopes.

Reviewer: ```suggestion
        - Under **Scopes**, add `dovecotprofile` and `authentik default OAuth Mapping: OpenID 'offline_access'` to the list of selected scopes.
```

I hope "Under Scopes" is correct, I did the name off memory. I feel like it's important to at least include this bit, the rest is a little wording improvement which fits what I'm used to correct PRs to

- **Configure Bindings** _(optional)_: you can create a [binding](/docs/add-secure-apps/flows-stages/bindings/) (policy, group, or user) to manage the listing and access to applications on a user's **My applications** page.

3. Click **Submit** to save the new application and provider.

## Roundcube configuration

Set the following variables in you Roundcube configuration file:

> ```sh

Reviewer: ```suggestion
```php
```

If I'm not mistaken, `$config` is used in PHP, so this syntax would be more appropriate. However, I did see a PR (https://github.com/goauthentik/authentik/pull/16683) which adds syntax highlighting to Powershell, it would be nice if all languages worked and we could use whichever in the codeblock, I didn't look too much into the issue you faced which required that PR.

```

:::note

> With this setup Dovecot can also be used with other email clients that support XOAUTH2 authentication, however most available software (including Fair Email for Android and Thunderbird) only come with support for Gmail, Outlook etc with no way to configure custom email servers.

Reviewer: ```suggestion
With this setup, Dovecot can also be used with other email clients that support XOAUTH2 authentication. However, most commonly available clients, such as FairEmail for Android and Thunderbird, only provide built-in support for providers like Gmail and Outlook, with no option to configure custom mail servers.
```
